genesyscloud/resource_genesyscloud_architect_user_prompt.go

```diff
@@ -0,0 +1,1081 @@
+			if resourceTtsString != nil || resourceTtsString.(string) != "" {
+				strResourceTtsString := resourceTtsString.(string)
+				promptResource.TtsString = &strResourceTtsString
+			}
+
+			resourceText := resourceMap["text"]
+			if resourceText != nil || resourceText.(string) != "" {
+				strResourceText := resourceText.(string)
+				promptResource.Text = &strResourceText
+			}
+
+			log.Printf("Creating user prompt resource for language: %s", resourceLanguage)
+			userPromptResource, _, err := architectApi.PostArchitectPromptResources(*userPrompt.Id, promptResource)
+			if err != nil {
+				return diag.Errorf("Failed to create user prompt resource %s: %s", name, err)
+			}
+			uploadUri := userPromptResource.UploadUri
+
+			resourceFilename := resourceMap["filename"]
+			if resourceFilename.(string) == "" {
+				continue
+			}
+			resourceFilenameStr := resourceFilename.(string)
+
+			status, content := uploadPrompt(uploadUri, &resourceFilenameStr, sdkConfig)
+			log.Printf("Status of upload: %s", status)
+			log.Printf("Content of upload: %s", content)
+		}
+	}
+
+	d.SetId(*userPrompt.Id)
+
+	log.Printf("Created user prompt %s %s", name, *userPrompt.Id)
+
+	return readUserPrompt(ctx, d, meta)
+}
+
+func readUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectAPI := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Reading User Prompt %s", d.Id())
+	userPrompt, resp, getErr := architectAPI.GetArchitectPrompt(d.Id())
+	if getErr != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Failed to read User Prompt %s: %s", d.Id(), getErr)
+	}
+
+	d.Set("name", *userPrompt.Name)
+	d.Set("description", *userPrompt.Description)
+
+	if userPrompt.Name != nil {
+		d.Set("name", *userPrompt.Name)
+	} else {
+		d.Set("name", nil)
+	}
+
+	if userPrompt.Description != nil {
+		d.Set("description", *userPrompt.Description)
+	} else {
+		d.Set("description", nil)
+	}
+
+	d.Set("resources", flattenPromptResources(userPrompt.Resources))
+
+	log.Printf("Read Audio Prompt %s %s", d.Id(), *userPrompt.Id)
+	return nil
+}
+
+func updateUserPrompt(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	architectApi := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Updating user prompt %s", name)
+	_, _, err := architectApi.PutArchitectPrompt(d.Id(), platformclientv2.Prompt{
+		Name:        &name,
+		Description: &description,
+	})
+	if err != nil {
+		return diag.Errorf("Failed to update user prompt %s: %s", name, err)
+	}
+
+	diagErr := updatePromptResource(d, architectApi, sdkConfig)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Printf("Updated User Prompt %s", d.Id())
```

Comment:
Also, a sleep of 5 seconds here so the read is guaranteed to return up to date data.